service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/DeleteFederatedCredentialStep.java

```diff
@@ -30,38 +29,37 @@ public DeleteFederatedCredentialStep(
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    if (!FlightUtils.getRequired(
-        context.getWorkingMap(), FEDERATED_IDENTITY_EXISTS, Boolean.class)) {
-      logger.info("Federated identity already gone");
-      return StepResult.getStepResultSuccess();
-    }
-
     final AzureCloudContext azureCloudContext =
         context
             .getWorkingMap()
             .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
     var msiManager = crlService.getMsiManager(azureCloudContext, azureConfig);
 
-    deleteFederatedCredentials(
-        msiManager,
-        k8sNamespace,
-        GetManagedIdentityStep.getManagedIdentityName(context),
-        k8sNamespace);
-    return StepResult.getStepResultSuccess();
-  }
-
-  private void deleteFederatedCredentials(
```

Comment:
I just inlined this function because it made the error harder to spot (passing in `k8sNamespace` instead of `azureCloudContext.getAzureResourceGroupId()`